great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py

```diff
@@ -1750,161 +2279,216 @@ def _filter(
 
         return column_based_expectation_configurations_by_type
 
+    def _filter_attributed_metrics_by_column_names(
+        self,
+        attributed_metrics: Dict[Domain, Dict[str, ParameterNode]],
+        include_column_names: Optional[List[str]],
+        exclude_column_names: Optional[List[str]],
+    ) -> Dict[Domain, Dict[str, ParameterNode]]:
+        def _filter(m: Dict[Domain, Dict[str, ParameterNode]]) -> bool:
+            column_name: str = m.domain_kwargs.column
+            if exclude_column_names and column_name in exclude_column_names:
+                return False
+            if include_column_names and column_name not in include_column_names:
+                return False
+            return True
+
+        domains: Set[Domain] = set(filter(lambda m: _filter(m), attributed_metrics))
+        filtered_attributed_metrics: Dict[Domain, Dict[str, ParameterNode]] = {
+            domain: attributed_metrics[domain] for domain in domains
+        }
+
+        return filtered_attributed_metrics
+
+    def _filter_attributed_metrics_by_metric_name(
+        self,
+        attributed_metrics: Dict[Domain, Dict[str, ParameterNode]],
+        metric_name: str,
+    ) -> Dict[Domain, Dict[str, ParameterNode]]:
+        domain: Set[Domain]
+        filtered_attributed_metrics: Dict[Domain, Dict[str, ParameterNode]] = {}
+        for domain, attributed_metric_values in attributed_metrics.items():
+            if metric_name in attributed_metric_values.keys():
+                filtered_attributed_metrics[domain] = {
+                    metric_name: attributed_metric_values[metric_name]
+                }
+
+        return filtered_attributed_metrics
+
     def _chart_domain_values(
         self,
+        expectation_type: str,
         df: pd.DataFrame,
         metric_name: str,
-        metric_type: alt.StandardType,
         plot_mode: PlotMode,
         sequential: bool,
         subtitle: Optional[str],
     ) -> Optional[alt.Chart]:
-        implemented_metrics: Set[str] = {
-            sanitize_parameter_name(metric)
-            for metric in self.EXPECTATION_METRIC_MAP.values()
+        nominal_metrics: Set[str] = {"table_columns"}
+
+        quantitative_metrics: Set[str] = {
+            "table_row_count",
+            "column_distinct_values_count",
+            "column_min",
+            "column_max",
+            "column_values_between",
+            "column_mean",
+            "column_median",
+            "column_quantile_values",
+            "column_standard_deviation",
```

Comment:
Any possibility of being able to take these strings and move them to an enum? Perhaps `PlotEnabledMetrics` or something? Could even be an internal enum nested within this class just so we have more security.

Reply:
I will look into doing this in the follow-up PR: https://github.com/great-expectations/great_expectations/pull/5259